### PR TITLE
Add add, clear completed and edit functionality

### PR DESCRIPTION
**Pull request by Sly-Cloud**
- [ ] Create a new JavaScript file for the new functionality.

- [ ] Implement a function for adding a new task

- [ ] Implement a function for editing task descriptions.

- [ ] Implement a function for deleting a task (remove an element from the array).

- [ ] Implement a function for the "Clear all completed" button (use filter() method).

By default new tasks should have the property completed set to false and th
Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
All changes to the To Do List should be saved in local storage.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello @Sly-Cloud 

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._

Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/add-remove.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// eslint-disable-next-line import/no-cycle
+import checkboxesEvent, {
+  list, save, fixIndex, setList, remove,
+} from './status-update';
+
+const todoList = document.getElementById('todo-list');
+export const items = document.getElementsByClassName('task-item');
```
Comment (mikenath223) [inline src/add-remove.js:7]: - [ ] It's not clear how to delete a single to-do item. Consider toggling the 3-dot-icon with the delete icon.

File: src/add-remove.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    editables[i].addEventListener('click', () => {
+      editables[i].parentNode.children[3].classList.add('show');
+      editables[i].parentNode.children[2].classList.add('hide');
+      editables[i].addEventListener('blur', (event) => {
+        if (event.target) {
+          localStorage.setItem('edit', JSON.stringify(editables[i].innerHTML));
+          list[i].description = JSON.parse(localStorage.getItem('edit'));
+          save();
+        }
+        // editables[i].parentNode.children[3].classList.remove('show')
```
Comment (mikenath223) [inline src/add-remove.js]: - [ ] Kindly remove this commented code to avoid making the codebase untidy.

File: src/add-remove.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+export function add() {
+  document.getElementById('task-entry').addEventListener('keypress', (event) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      const description = document.getElementById('task-entry');
+      const id = list.length;
+      const task = { description: description.value, completed: false, index: id };
+      description.value = '';
+      list.push(task);
+      save();
```
Comment (mikenath223) [inline src/add-remove.js]: - [ ] Please change these and all other regular functions to es6 arrow function syntax.

File: src/status-update.js [hunk 4]
```diff
@@ -1,7 +1,34 @@
-import { list, save } from './storage';
+// eslint-disable-next-line import/no-cycle
+import { displayTasks, items } from './add-remove';
+// eslint-disable-next-line import/no-mutable-exports
```
Comment (mikenath223) [inline src/status-update.js]: - [ ] Also remove these commented codeblocks.

File: src/add-remove.js (hunk 1 above)
Comment (Sly-Cloud) [reply src/add-remove.js:7]: Alright, Thanks for the review

File: src/add-remove.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
+// eslint-disable-next-line import/no-cycle
+import checkboxesEvent, {
+  list, save, fixIndex, setList, remove,
+} from './status-update';
+
+const todoList = document.getElementById('todo-list');
+export const items = document.getElementsByClassName('task-item');
```
Comment (rloterh) [inline src/add-remove.js:7]: - [ ] Nice job for adding the _trash icon_ for deleting each task. However, the _trash icon_ **does not remove** an item when clicked on. Kindly ensure to improve upon this aspect.

**Your design:**
image

File: src/status-update.js (hunk 4 above)
Comment (rloterh) [reply src/status-update.js]: This issue is not fully resolved. Kindly endeavor to remove **all** the above code blocks within your code, so as to make your code more robust and cleaner.

File: src/add-remove.js (hunk 3 above)
Comment (rloterh) [reply src/add-remove.js]: This issue is not fully resolved.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Sly, 

You did great on this project, and for making the changes requested by the previous reviewer! 👏👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- No linter errors

## Required Changes ♻️
_Kindly check the comments above, made under the reviews of the previous code reviewer(s) and stating **not properly resolved**. Kindly take your time to correctly review those issues outlined by the previous code reviewers, and please check to see that the changes are working as intended._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Sly-Cloud** [comment]
I have done all the things I could but I can't seem to work around a solution on the ESLint comment issue. It creates an error when I remove them. I need more time than what I have now, and this would be my last submission. Please consider this, as all other core features are working. Thanks

File: src/status-update.js (hunk 4 above)
Comment (vichuge) [reply src/status-update.js]: These commented lines are disabled for eslint rules, 1 or 2 is ok in a script.

**Review by vichuge — APPROVED** [review]
# Approved project! ✅ 🥳 🎈 🎉🎉🎉🎉🎉🎉🎉🎉 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

InbornCandidHornet-size_restricted

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**